tasks/generate-history/registerArchaeologists.ts

```diff
@@ -0,0 +1,65 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { BigNumber } from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { generateHistoryConfig } from "./config";
+import { range } from "./helpers";
+
+/**
+ * Gives an archaeologist sarco token, approves the sarco token, and registers the archaeologist.
+ * Error stack traces are left out to keep the logs clean.
+ * @param hre The Hardhat Runtime Environment
+ */
+export async function registerArchaeologists(
+  hre: HardhatRuntimeEnvironment
+): Promise<SignerWithAddress[]> {
+  const accounts = await hre.ethers.getSigners();
+  const count = generateHistoryConfig.archaeologistCount;
+
+  if (count > accounts.length) {
+    throw new Error(`Not enough accounts. Count must be less than or equal to ${accounts.length}`);
+  }
+
+  // Get the contract
+  const diamond = await hre.ethers.getContract("Diamond_DiamondProxy");
+  const archaeologistFacet = await hre.ethers.getContractAt("ArchaeologistFacet", diamond.address);
+  const sarcoToken = await hre.ethers.getContract("SarcoTokenMock");
+
+  console.log("Registering archaeologists...");
+  const archaeologistSigners: SignerWithAddress[] = [];
+  for (let i = 0; i < count; i++) {
+    const account = accounts[i];
+    const minimumDiggingFee = 10;
+
+    // 1 week
+    const maximumRewrapInterval = 604800;
+
+    // Be sure to provide plenty of free bond for the archaeologist
+    const freeBond = 10000;
+
+    // Register the archaeologist
+    try {
+      await sarcoToken.connect(account).approve(diamond.address, hre.ethers.constants.MaxUint256);
+      await sarcoToken.transfer(account.address, freeBond * 10);
+      await archaeologistFacet
+        .connect(account)
+        .registerArchaeologist(
+          "some-peer-id",
```

Comment:
If we wanted to be able to hook up real archaeologists to these accounts (which may be useful), this could generate a real peerId. We would probably need to output the full peer Id JSON + signer private key (either via logs or write to a file) so archs could be hooked up with the registering account.

Otherwise we would need to disable the "check if online" portion of the front-end to see the archs, as they won't show up, which maybe what we want to do at this point to save time (we could discuss this).